### PR TITLE
feat: allow to export logs

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

**What type of PR is this?**

/kind feature
/kind cleanup
/kind api-change

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
Search API has performance issue, use Scroll API instead
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/logging/v1alpha2/register.go [hunk 1]
```diff
@@ -71,6 +71,28 @@ func addWebService(c *restful.Container) error {
 		Consumes(restful.MIME_JSON, restful.MIME_XML).
 		Produces(restful.MIME_JSON)
 
+	ws.Route(ws.GET("/cluster/export").To(logging.LoggingClusterExport).
```
Comment (zryfish) [inline pkg/apis/logging/v1alpha2/register.go]: export is a verb, not suitable. I think logs would be good

File: pkg/apiserver/logging/logging.go [hunk 2]
```diff
@@ -166,13 +166,14 @@ func logExport(param v1alpha2.QueryParameters, response *restful.Response) {
...
-	// retrieve 1000 records each time
+	// export 1000 records in every iteration
 	param.Size = 1000
 	// from is not allowed in a scroll context
 	param.From = 0
 
 	var scrollId string
-	for {
+	// limit to retrieve max 100k records
+	for i := 0; i < 100; i++ {
```
Comment (benjaminhuo) [inline pkg/apiserver/logging/logging.go]: 这个 100 是不是单独作为一个 const 变量好点？

Comment (huanggze) [reply pkg/apiserver/logging/logging.go]: 暂时只有这里有这样的限制，直接写限制 100 次循环应该也可以

File: pkg/apiserver/logging/logging.go [hunk 3]
```diff
@@ -188,22 +139,86 @@ func LoggingDeleteFluentbitOutput(request *restful.Request, response *restful.Re
...
+	// export 1000 records in every iteration
+	param.Size = 1000
+	// from is not allowed in a scroll context
+	param.From = 0
+
+	var scrollId string
+	// limit to retrieve max 100k records
+	for i := 0; i < 100; i++ {
+		var res *v1alpha2.QueryResult
+		if scrollId == "" {
```
Comment (zryfish) [inline pkg/apiserver/logging/logging.go]: What if user cancelled request in the middle of downloading? I believe this loop will continue. Need to break loop quickly if user cancelled downloading. You can try this https://gist.github.com/zryfish/353dddb2f7eb4dd99e195b0270d52e9b

Comment (huanggze) [reply pkg/apiserver/logging/logging.go]: we interrupt downloading by capture error from io.Copy

Comment (zryfish) [reply pkg/apiserver/logging/logging.go]: no, that's not right. Even request is cancelled, response is still there, keep writing. I modified the example, you can try

**Pull request merged**